test/e2e/tests/test_references.py

```diff
@@ -32,28 +34,42 @@
 # Little longer to delete the policy since it's referred-to from the role...
 DELETE_POLICY_TIMEOUT_SECONDS = 30
 CHECK_WAIT_AFTER_REF_RESOLVE_SECONDS = 10
+TESTING_NAMESPACE = "custom_namespace"
+
+log = getLogger(__name__)
 
 @pytest.fixture(scope="module")
 def referred_policy_name():
     return random_suffix_name("referred-policy", 24)
 
 
 @pytest.fixture(scope="module")
-def referring_role(referred_policy_name):
+def referring_role(request, referred_policy_name):
     role_name = random_suffix_name("referring-role", 24)
 
+    marker = request.node.get_closest_marker("resource_data")
+    filename = "role_referring"
+    namespace = "default"
+
+    if marker is not None:
+        data = marker.args[0]
+        if 'withNamespace' in data:
```

Comment:
Should also investigate `data['withNamespace']` ? maybe it's set to 'false'/False?